specs/emio-006.md

```diff
@@ -0,0 +1,104 @@
+2) **Moral incentives** (who is following the protocol, contributing to the community, etc.)
+
+Given that
+
+1) There are hundreds or thousands of pools to choose from
+2) *Financial incentives* are very quick to compare
+3) *Moral incentives* are very hard to compare
+
+Most users will simply choose their pool based on the *financial incentive*.
+
+This is problematic because:
+
+1) This is sometimes detrimental to the protocol (an adversary could promise high returns but attack the protocol)
+2) Empirically many users want to vote based on their morals but were not able to.
+
+Therefore we need a way to simplify choosing based on **moral incentives** without resulting in centralization.
+
+# Proposal
+
+We propose to use a standard format for encoding pool curation information. A standard format allows us to decouple software creators from curators. For example, a wallet or explorer could bundle their software with a pre-existing set of curators but also allow users to override to use their own preferred curator.
+
+### Format
+
+We propose the format follows (defined using [JSON Schema](https://json-schema.org/)).
+
+```
+{
+  "definitions": {
+    "node_flags": {
+      "$id": "#/definitions/node_flags",
+      "type": "integer",
+      "title": "Node-level Flags",
+      "description": "Flags that can be set by node software indicating automatically-detected adversarial behavior",
+      "default": 0
+     }
+  },
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "patternProperties": {
+    "^[a-f0-9]{64}$": {
+      "type": "object",
+      "properties": {
+        "node_flags": { "$ref": "#/definitions/node_flags" }
+      },
+      "additionalProperties": false
+    }
+  },
+  "additionalProperties": false
+}
+```
+
+Here is an example
+
+```json
+{
+  "0000111122223333444455556666777788889999aaaabbbbccccddddeeeeffff": {
+    "node_flags": 1
+  },
+  "ffffeeeeddddccccbbbbaaaa9999888877776666555544443333222211110000": {
+    "node_flags": 3
+  },
+}
+```
+
+Note: keys are *pool ids* (32-byte hex strings). Since keys are unique, it multiple curation systems can provide data for the same pools and the results can easily be merged together.
+
+### Security considerations
+
+The result of the endpoint should NOT be parsed as any kind of code that could be executed.
+
+### Field registry
+
+Here we provide an extensible list of keys used for this specification
+
+##### node_flags
+
+```json
+"node_flags": {
+  "$id": "#/definitions/node_flags",
+  "type": "integer",
+  "title": "Node-level Flags",
+  "description": "Flags that can be set by node software indicating automatically-detected adversarial behavior",
+  "default": 0
+}
+```
+
+This is a bitset with the following table used for values
+
+| value | meaning |
+|---|---|
+| 0 | no adversarial behavior detected (prefer to omit the field entirely in this case) |
+| 1 | multiple blocks for a single slot |
+| 2 | censoring transactions (ex: no transactions in blocks) |
```

Comment:
🤔 Any way to prove the censoring reliably?

Reply:
I leave it up to whatever curator to decide how to do this. Clearly if a pool is never including transactions in their blocks then something is going on. It can get more suspicious with Goguen where maybe a pool is censoring transactions to a smart contract. Some people can try and come up with heuristics for this but it's outside of the scope of this spec.

Reply:
Makes sense. Thinking if we need to add another field for an arbitrary comment or a link to the reasoning.

Reply:
I agree with having an arbitrary field to identify reasoning - as we can't be certain of all adversarial vectors we'll see an arbitrary field will help address that.  I'm thinking perhaps such a list might be best curated by a council of stakeholders to aid in establishing/maintaining best practices relevant to sustaining reliable censorship.